app/models/manageiq/providers/embedded_ansible/automation_manager/machine_credential.rb

```diff
@@ -94,6 +94,8 @@ class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::MachineCredential
   alias ssh_key_data   auth_key
   alias ssh_key_unlock auth_key_password
 
+  before_validation :ensure_newline_for_ssh_key
```

Comment:
why not have this in the base credential as well?  I can't imagine a scenario where we _wouldn't_ want this fix for any auth_key in any subclass.

Reply:
@Fryguy because sometimes that isn't used for an SSH key, but things like Amazon's IAM credentials, so I really want to be targeted where I am putting this change.